File: 7-days-of-code/array.js
class ListaDeCompras{
    itens = [];
    quantidades = [];

    adicionar_item(nome_produto, quantidade){
        if(nome_produto == ''){
            console.log("Você precisa digitar o nome e quantidade do produto");
            return false;
        }

        this.itens.push(nome_produto);
        this.quantidades.push(quantidade);
    }

    remover_item(nome_produto){
        const produto_index = this.itens.indexOf(nome_produto);

        if(nome_produto == ''){
            console.log('Você precisa digitar o nome do produto');
            return false;
        }

        if(produto_index == -1){
            console.log('Produto não encontrado');
            return false;
        }

        this.itens.splice(produto_index, 1);
        this.quantidades.splice(produto_index, 1);
    }

    listar_itens(){
        for(let i = 0; i < this.itens.length; i++){
            console.log(`[PRODUTO]: ${this.itens[i]} - [QUANTIDADE]: ${this.quantidades[i]}`)
        }
    }

}


const lista_de_compras_natalia = new ListaDeCompras();

lista_de_compras_natalia.adicionar_item('maça', 4);
lista_de_compras_natalia.adicionar_item('arroz', 3);
lista_de_compras_natalia.adicionar_item('tortinha de brigadeiro', 10);
lista_de_compras_natalia.adicionar_item('leite integral', 12);
lista_de_compras_natalia.adicionar_item('', 2);

lista_de_compras_natalia.remover_item('leite integral');
lista_de_compras_natalia.remover_item('chocolate');
lista_de_compras_natalia.listar_itens();

File: 7-days-of-code/python/array.py
class ListaDeCompras: 
    def __init__(self):
        self.items = []
        self.quantidade = []

    def adicionar_produto(self, nome, quantidade=None):
        if(nome == ''):
            print("Você precisa digitar o nome do produto")
            return False
        
        if(quantidade == None):
            self.quantidade.append(1)
        else:
            self.quantidade.append(quantidade)
        
        self.items.append(nome)
    
    def remove_produto(self, nome):
        if(nome == ''):
            print('Você precisa digitar o nome do produto')
            return 
        if nome not in self.items:
            print('Produto não encontrado')
            return

        indexProduto = self.items.index(nome)  
        self.items.pop(indexProduto)
        self.quantidade.pop(indexProduto)
    
    def exibirLista(self):
        for x in range(len(self.items)):
            print('PRODUTO:', self.items[x], '|', 'QUANTIDADE:', self.quantidade[x])


# LISTA DE COMPRAS
lista_123 = ListaDeCompras()

# ADICIONAR PRODUTOS
lista_123.adicionar_produto('Sabonete', 3)
lista_123.adicionar_produto('Shampoo')
lista_123.adicionar_produto('Arroz', 2)

lista_123.exibirLista();

# REMOVER PRODUTO
lista_123.remove_produto('Shampoo')

lista_123.exibirLista();